caas/kubernetes/provider/cloud.go

```diff
@@ -306,93 +306,36 @@ func (p kubernetesEnvironProvider) FinalizeCloud(ctx environs.FinalizeCloudConte
 	return cld, nil
 }
 
-func checkMicrok8sUserGroupSetup(cmdRunner CommandRunner) error {
-	if jujuos.HostOS() == jujuos.Windows || jujuos.HostOS() == jujuos.OSX {
-		// The microk8s on windows and macOS is running on a vm managed by multipass.
-		// Even the vm does not have the user group properly configured but it is not
-		// a problem because microk8s CLI uses `multipass exec` with sudo to run the commands.
-		// https://github.com/ubuntu/microk8s/blob/master/installer/vm_providers/_multipass/_multipass.py#L50
-		return nil
-	}
-	resp, err := cmdRunner.RunCommands(exec.RunParams{
-		Commands: `id -nG "$(whoami)" | grep -qw "root\|microk8s"`,
-	})
-	if err != nil {
-		return errors.Annotate(err, "checking microk8s setup")
+func checkDefaultStorageExist(broker ClusterMetadataStorageChecker) error {
+	storageClasses, err := broker.ListStorageClasses(k8slabels.NewSelector())
+	if err != nil && !errors.IsNotFound(err) {
+		return errors.Annotate(err, "cannot list storage classes")
 	}
-	if resp.Code != 0 {
-		user, _ := utils.OSUsername()
-		if user == "" {
-			user = "<username>"
+	for _, sc := range storageClasses {
+		if sc.Annotations["storageclass.kubernetes.io/is-default-class"] == "true" {
+			return nil
 		}
-		return errors.Errorf(`
-The microk8s user group is created during the microk8s snap installation.
-Users in that group are granted access to microk8s commands and this
-is needed for Juju to be able to interact with microk8s.
-
-Add yourself to that group before trying again:
-	sudo usermod -a -G microk8s %s
-`[1:], user)
 	}
-	return nil
+	return errors.New("required storage addon is not enabled")
```

Comment:
"no default storage is defined"

Reply:
I would prefer the above error message which is more accurate and descriptive?

Reply:
This is a generic message though. "storage addon" is a microk8s specific thing.
The method is "checkDefaultStorageExists" and it simply looks for a default storage class, which is generic to nay k8s, hence the error message also needs to be applicable anywhere.